src/clj/puppetlabs/puppetserver/ringutils.clj

```diff
@@ -85,6 +109,28 @@
         false))
     true))
 
+(schema/defn ^:always-validate request->name :- (schema/maybe schema/Str)
+  "Return the name of the client by parsing the subject of the client cert from
+  the request.  Return nil if the client cert is absent or cannot be parsed."
+  [request :- RingRequest]
+  (let [{:keys [ssl-client-cert]} request]
+    (if ssl-client-cert (get-cert-subject ssl-client-cert))))
+
+(schema/defn ^:always-validate authorized? :- schema/Bool
+  "(SERVER-768) replacement for client-allowed-access?"
+  [rules :- rules/Rules
+   request :- RingRequest]
+  (if-let [name (request->name request)]
```

Comment:
I spent a few minutes thinking about this...  My thinking was that an authorization rule that allows all clients still makes the assumption that the client is authenticated.  This led me to think a client still needs at least a client cert given the current design of trapperkeeper-authorization.

With that said, it seems like we'll need _some_ way to handle SSL clients that don't have a certificate and as such don't have a name.  I think this should be a change to the upstream authorization library as we need some first-class way to say, "this client is authentic, it's just authentic without any way to identify it, so it has no name."

I'd hate to overload something like nil or the empty string, but it may not be so terrible to do so.